# backtest-case: CXX-cublaslt_grouped_gemm
# kind: cxx
# base: 42b840051647eef89761a16dfdff87e82bb253ab
# derived: diff(BASE:transformer_engine/common/gemm/cublaslt_grouped_gemm.cu, PRE:transformer_engine/common/gemm/cublaslt_grouped_gemm.cu)
--- a/transformer_engine/common/gemm/cublaslt_grouped_gemm.cu
+++ b/transformer_engine/common/gemm/cublaslt_grouped_gemm.cu
@@ -1,4 +1,6 @@
 /*************************************************************************
+* This file was modified for portability to AMDGPU
+* Copyright (c) 2025-2026, Advanced Micro Devices, Inc. All rights reserved.
 * Copyright (c) 2022-2026, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 *
 * See LICENSE for license information.
@@ -21,6 +23,8 @@
 #include "../util/vectorized_pointwise.h"
 #include "./config.h"
 
+#ifndef __HIP_PLATFORM_AMD__
+
 namespace {
 
 inline void CreateCublasHandle(cublasLtHandle_t *handle) {
@@ -1407,6 +1411,46 @@
 
 #endif  // CUBLAS_VERSION >= CUBLAS_GROUPED_GEMM_VERSION
 
+#else //__HIP_PLATFORM_AMD__
+
+void nvte_grouped_gemm(const NVTEGroupedTensor A, int transa, const NVTEGroupedTensor B, int transb,
+                       const NVTEGroupedTensor C, NVTEGroupedTensor D, const NVTETensor alpha,
+                       const NVTETensor beta, NVTETensor workspace_setup,
+                       NVTETensor workspace_cublas, NVTEGroupedMatmulConfig config,
+                       cudaStream_t stream) {
+  NVTE_ERROR("nvte_grouped_gemm is not supported on ROCm yet");
+}
+
+void nvte_grouped_gemm_with_discrete_inputA(const NVTETensor *A_list, size_t num_a_tensors,
+                                            int transa, const NVTEGroupedTensor B, int transb,
+                                            const NVTEGroupedTensor C, NVTEGroupedTensor D,
+                                            const NVTETensor alpha, const NVTETensor beta,
+                                            NVTETensor workspace_setup, NVTETensor workspace_cublas,
+                                            NVTEGroupedMatmulConfig config, cudaStream_t stream) {
+  NVTE_ERROR("nvte_grouped_gemm_with_discrete_inputA is not supported on ROCm yet");
+}
+
+void nvte_grouped_gemm_with_discrete_out(const NVTEGroupedTensor A, int transa,
+                                         const NVTEGroupedTensor B, int transb,
+                                         const NVTETensor *C_list, size_t num_c_tensors,
+                                         NVTETensor *D_list, size_t num_d_tensors,
+                                         const NVTETensor alpha, const NVTETensor beta,
+                                         NVTETensor workspace_setup, NVTETensor workspace_cublas,
+                                         NVTEGroupedMatmulConfig config, cudaStream_t stream) {
+  NVTE_ERROR("nvte_grouped_gemm_with_discrete_out is not supported on ROCm yet");
+}
+
+void nvte_grouped_bias_add(const NVTEGroupedTensor output, const NVTEGroupedTensor bias,
+                           cudaStream_t stream) {
+  NVTE_ERROR("nvte_grouped_bias_add is not supported on ROCm yet");
+}
+
+size_t nvte_get_grouped_gemm_setup_workspace_size(size_t num_tensors) {
+  NVTE_ERROR("nvte_get_grouped_gemm_setup_workspace_size is not supported on ROCm yet");
+  return 0;
+}
+
+#endif // __HIP_PLATFORM_AMD__
 namespace {
 
 __global__ void convert_int32_to_int64_kernel(const int32_t *src, int64_t *dst, size_t n) {
